app/javascript/packs/document-capture.tsx

```diff
@@ -67,8 +70,13 @@ function getMetaContent(name): string | null {
 const device: DeviceContextValue = { isMobile: isCameraCapableMobile() };
 
 const trackEvent: typeof baseTrackEvent = (event, payload) => {
-  const { flowPath } = appRoot.dataset;
-  return baseTrackEvent(event, { ...payload, flow_path: flowPath });
+  const { flowPath, acuantSdkUpgradeABTestingEnabled, useNewerSdk } = appRoot.dataset;
+  return baseTrackEvent(event, {
+    ...payload,
+    flow_path: flowPath,
+    acuant_sdk_upgrade_a_b_testing_enabled: acuantSdkUpgradeABTestingEnabled,
+    use_newer_sdk: useNewerSdk,
```

Comment:
Should we consider populating the context with the Acuant version, and using that as part of the analytics event, as well as in place of the constants in `acuant-sdk-upgrade-a-b-test.tsx`? Or, maybe we don't even need an extra context provider, and can just use `AcuantContext` in the way it was designed to receive the `sdkSrc`?

(Short on time to elaborate at the moment, but will follow-up with more detail on what I had in mind)

Reply:
What I was thinking is maybe the frontend would provide an SDK version if and only if A/B testing is enabled, and then that version (if given) is used to derive a source string to be given directly to the existing `AcuantContext`, so that we don't need a new context, and so that `AcuantContext` wouldn't require many (or any) additional changes.

In [`packs/document-capture.tsx`](https://github.com/18F/identity-idp/blob/04ec6ca2fb22235c714fb2e29861849e3d168d07/app/javascript/packs/document-capture.tsx#L106-L133)

```ts
interface AppRootData {
  // ...
  acuantVersion?: string;
}

const {
  // ...
  acuantVersion,
} = appRoot.dataset;

const isAcuantSDKUpgradeABTestingEnabled = typeof acuantVersion === 'string';

const trackEvent: typeof baseTrackEvent = (event, payload) => {
  // ...
  return baseTrackEvent(event, {
    ...payload,
    // ...
    acuant_sdk_upgrade_a_b_testing_enabled: isAcuantSDKUpgradeABTestingEnabled,
    acuant_version: acuantVersion,
  });
};

const App = composeComponents(
  // ...
  [
    AcuantContextProvider,
    {
      // ...
      sdkSrc: acuantVersion && `/acuant/${acuantVersion}/AcuantJavascriptWebSdk.min.js`,
```

Reply:
Makes sense. The one amendment I would suggest is to also have the actual `acuant_sdk_upgrade_a_b_testing_enabled` value also be on the AppRoot, because this way we can specify the acuant version to use regardless of the AB test (ie, let's always specify a version to use). Otherwise we would have hard coded values for the version still present in the acuant context

Reply:
I just pushed [6c1a5e1](https://github.com/18F/identity-idp/pull/7392/commits/6c1a5e1bdb25685ce433ccef36ba4486741f689a) which should implement most of your suggestions, along with some minor differences. 

To me the outstanding issues are -- as I mentioned in the commit message -- whether/where to store the version numbers as globally accessible values, and whether or not we track an event for when the acuant sdk successfully loads or not.

Reply:
>whether/where to store the version numbers as globally accessible values

Do we need to access them anywhere else? If not, I don't think we'd need to do anything.

If we do, we can probably expose / access `AcuantConfig.acuantVersion` which would give us the version detail after the script is loaded.

>whether or not we track an event for when the acuant sdk successfully loads or not.

What questions are we trying to answer with such an event? Are we concerned that the scripts might not load correctly?

Could we indirectly infer that from this event?

https://github.com/18F/identity-idp/blob/8c40e370f935871ab995113716d14b1e209ccbc1/app/javascript/packages/document-capture/context/acuant.tsx#L253-L256

We could add the `AcuantConfig.acuantVersion` in there if we wanted?

Reply:
> Could we indirectly infer that from this event?
  
Yeah for sure, since we wrapped the default `trackEvent` in document capture and add all that extra information (in the pack). I am not so savvy on Cloudwatch queries so I don't know how much of a pain that is, however. If you think it's GTG let's just roll with it.
  
> Do we need to access them anywhere else?
  
I am only thinking of the version numbers present in the default paths of the AcuantContext provider [here](https://github.com/18F/identity-idp/blob/eric-lg-8190/app/javascript/packages/document-capture/context/acuant.tsx#L196). All other version information is defined on the Rails side now.

Reply:
Oh, right, I forgot we added it in the base `trackEvent`, so it should be easy to distinguish in CloudWatch by e.g. `filter properties.event_properties.acuant_version = '11.7.0'`, etc.